src/main/java/step2/Ball.java

```diff
@@ -0,0 +1,45 @@
+package step2;
+
+import java.util.Objects;
+
+public class Ball implements Comparable<Ball>{
+    public static final String VALID_BALL_NUMBER_MSG = "로또공의 숫자는 1부터 45사이여야 합니다.";
+    private final int number;
+
+    public Ball(int number) {
+        if (number < 1 || number > 45) {
+            throw new IllegalArgumentException(VALID_BALL_NUMBER_MSG);
+        }
+        this.number = number;
+    }
+
+    public Ball(String number) {
+        this(Integer.parseInt(number.trim()));
+    }
+    public Ball(Ball ball) {
+        this.number = ball.number;
+    }
+
+    @Override
+    public String toString() {
+        return number + "";
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (!(o instanceof Ball)) return false;
+        Ball ball = (Ball) o;
+        return number == ball.number;
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(number);
+    }
+
+    @Override
+    public int compareTo(Ball o) {
+        return new Integer(number).compareTo(o.number);
```

Comment:
Integer 객체를 생성하기보단 정적 메서드를 활용해보시면 어떨까요?
```suggestion
        return Integer.compare(number, o.number);
```


클래스 구현 순서는 다음과 같이 중요한 순서로 위치하는게 좋을 것 같아요!
참고 부탁드립니다

```
class  T {
    상수(static final) 또는 클래스 변수
    인스턴스 변수
    생성자
    팩토리 메소드
    메소드
    기본 메소드 (equals, hashCode, toString)
}
```